model/MediaSource.php

```diff
@@ -366,4 +341,61 @@ private function getFileSourceUnserializer(): FileSourceUnserializer
     {
         return $this->getServiceLocator()->get(FileSourceUnserializer::class);
     }
+
+    private function searchDirectories(
+        string $parentLink = '',
+        array $acceptableMime = [],
+        int $depth = 1,
+        int $childrenLimit = 0,
+        int $childrenOffset = 0
+    ): array {
+
+        $class = $this->getClass($parentLink == '' ? $this->getRootClassUri() : tao_helpers_Uri::decode($parentLink));
+
+        $data = [
+            'path' => self::SCHEME_NAME . tao_helpers_Uri::encode($class->getUri()),
+            'label' => $class->getLabel(),
+            'childrenLimit' => $childrenLimit,
+        ];
+
+        if ($depth > 0) {
+            $children = [];
+            foreach ($class->getSubClasses() as $subclass) {
+                $children[] = $this->searchDirectories(
+                    $subclass->getUri(),
+                    $acceptableMime,
+                    $depth - 1,
+                    $childrenLimit,
+                    $childrenOffset);
+            }
+
+            $filter = [];
+
+            if (!empty($acceptableMime)) {
+                $filter = array_merge($filter, [MediaService::PROPERTY_MIME_TYPE => $acceptableMime]);
+            }
+
+            $options = array_filter([
+                'limit' => $childrenLimit,
+                'offset' => $childrenOffset,
+            ]);
+
+            foreach ($class->searchInstances($filter, $options) as $instance) {
+                try {
+                    $children[] = $this->getFileInfo($instance->getUri());
+                } catch (tao_models_classes_FileNotFoundException $e) {
+                    $this->logEmergency(sprintf('Encountered issues "%s" while fetching details for %s',
+                            $e->getMessage(),
+                            $instance->getUri())
+                    );
```

Comment:
By reading it seems `$e->getMessage()` and `$instance->getUri()` are parameters of `logEmergency`, but they are not. It is just to bring some comfort reading and more cohesion in the indentation. What do you think? 

```suggestion
                    $this->logEmergency(
                        sprintf(
                            'Encountered issues "%s" while fetching details for %s',
                            $e->getMessage(),
                            $instance->getUri()
                        )
                    );
```